server/homebrew.api.js

```diff
@@ -81,6 +81,24 @@ const updateBrew = (req, res)=>{
 	HomebrewModel.get({ editId: req.params.id })
 		.then((brew)=>{
 			const updateBrew = excludePropsFromUpdate(req.body);
+
+			if(brew.authors.length > 0 && !(!!req.account && brew.authors.includes(req.account?.username))) {
+				console.warn(`User ${req.account?.username} is not an author on brew ${brew._id}`);
+				res.setHeader('Content-Type', 'application/json');
+				return res.status(403).send(JSON.stringify({ message: `You must be added as an author by the document owner to edit this brew!` }));
+			}
+
+			if(brew.version > updateBrew.version) {
+				console.warn(`Document ${brew._id} had a version mismatch`);
+				res.setHeader('Content-Type', 'application/json');
+				return res.status(409).send(JSON.stringify({ message: `The brew has been changed on a different device. Please save your changes elsewhere, refresh, and try again.` }));
+			}
+
+			if(req.account) {
+				brew.authors = _.uniq(_.concat([!!brew.authors ? brew.authors[0] : undefined], updateBrew.authors, req.account.username)).filter((a)=>a !== undefined && a.length > 0);
```

Comment:
If you are already filtering out undefined anyway, do we need `!!brew.authors ? brew.authors[0] : undefined` ? Can we reduce to `_.concat([brew?.authors?.[0]], updateBrew.authors, req.account.username)` ?